runtime/ops/tty.rs

```diff
@@ -270,10 +270,20 @@ fn console_size_from_fd(
     {
       return Err(Error::last_os_error());
     }
-    Ok(ConsoleSize {
-      cols: bufinfo.dwSize.X as u32,
-      rows: bufinfo.dwSize.Y as u32,
-    })
+
+    // calculate the size of the visible window
+    // * use over/under-flow protections b/c MSDN docs only imply that srWindow components are all non-negative
+    // * ref: <https://docs.microsoft.com/en-us/windows/console/console-screen-buffer-info-str> @@ <https://archive.is/sfjnm>
+    let cols = std::cmp::max(
+      bufinfo.srWindow.Right as i32 - bufinfo.srWindow.Left as i32 + 1,
```

Comment:
We can copy libuv here. width can be `dwSize.X`

https://github.com/libuv/libuv/blob/d2d92b74a8327daf9652a9732454937f3529bd30/src/win/tty.c#L175-L176

Reply:
I don't see any explanation in `libuv` for using the screen buffer width instead of the calculation from the console window view, so I'm not sure why they are using that assignment instead of doing the simple calculation.

I haven't seen any examples, but I don't know of any reason that the screen buffer width and the console window view width would always have to be equal. So, using `dwSize.X` seems to be falling into a future error similar to the height calculation that this PR fixes.